test/DynamoCoreWpfTests/CoreUITests.cs

```diff
@@ -966,12 +966,53 @@ public void InCanvasSearchTextChangeTriggersOneSearchCommand()
             DispatcherUtil.DoEvents();
 
             int count = 0;
-            (searchControl.DataContext as SearchViewModel).SearchCommand = new Dynamo.UI.Commands.DelegateCommand((object _) => { count++; });
+            var vm = searchControl.DataContext as SearchViewModel;
+            Assert.IsNotNull(vm);
+            vm.SearchCommand = new Dynamo.UI.Commands.DelegateCommand((object _) => { count++; });
+
+            // run without debouncer
+            vm.searchDelayTimeout = -1;
+            searchControl.SearchTextBox.Text = "dsfdf";
+            DispatcherUtil.DoEvents();
+
+            Assert.IsTrue(currentWs.InCanvasSearchBar.IsOpen);
+            Assert.AreEqual(1, count, "changing the text once should cause a single update");
+
+            // prepare debounce tests
+            var debounceTime = 10;
+            vm.searchDelayTimeout = debounceTime;
+
+            // run with debouncer
+            count = 0;
+            searchControl.SearchTextBox.Text = "dsfdfdsfdf";
+            Thread.Sleep(vm.searchDelayTimeout * 2);
```

Comment:
Are Thread.Sleep()'s okay during tests? I'm not sure how else to verify that the debounces work as expected.

Reply:
I think it's acceptable since these are relatively small. - I guess you could also make the delay time even smaller for testing though too small and it's probably not reliable anyway.